proxy/src/telemetry/metrics/prometheus.rs

```diff
@@ -34,8 +35,13 @@ struct Metric<M, L: Hash + Eq> {
     values: IndexMap<L, M>
 }
 
+/// A Prometheus counter is represented by a `Wrapping` unsigned 64-bit int.
+///
+/// Counters handle overflows by wrapping rather than saturating (or panicking!)
+/// since most timeseries systems (like Prometheus) are designed to handle this
+/// case gracefully.
```

Comment:
Please include a link to the Prometheus docs that say that wrapping is OK. I expected to find it at https://prometheus.io/docs/concepts/metric_types/#counter but it wasn't there. In fact, all I could find in discussions regarding Prometheus is that they don't worry about overflow because their counters use floating point, not integer, types, so they can handle very large magnitude. And that implies to me, in particular, that wrapping isn't OK.

Reply:
Yeah, I also attempted to find some discussion of wrapping behaviour in the Prometheus documentation, but I was also unable to. @olix0r, could you possibly refer us to some documentation regarding this?

Reply:
This is still unaddressed. The new comment seems to be wrong based on the investigation that we did so far: the correct semantics are to ensure the counter always gets reset to zero after Prometheus reads it, before it would ever overflow a 52-bit f64 mantissa. Note that we implement explicit wrapping to that release builds and debug builds have the same, not-quite-correct, semantics. Let's include the links that Oliver found for `irate()` and `resets()` that explain this.


1. Let's link to the page that Oliver linked to that talks about reset semantics.
2. Let's add a note that wrapping isn't exactly the right thing to do, but it is better than panicing.
3. Let's add `// TODO: Implement Prometheus reset semantics correctly, taking into consideration that prometheus models counters as `f64` and so there are only 52 significant bits`

Reply:
Yeah, I have yet to update the documentation but indended to before I considered this merge-ready; sorry!